### PR TITLE
[DROOLS-1438] Make property reactivity to work even when the property…

### PR DESCRIPTION
**Pull request by mariofusco**
… setter is in an inherited class

### CODE REVIEW

**Comment by mariofusco** [comment]
ok to test

File: drools-core/src/main/java/org/drools/core/util/ClassUtils.java [hunk 1]
```diff
@@ -411,13 +411,13 @@ public static String getPackage(Class<?> cls) {
...
-    public static List<String> getSettableProperties(Class<?> clazz) {
-        Set<SetterInClass> props = new TreeSet<SetterInClass>();
+    public static List<String> getAccessibleProperties( Class<?> clazz ) {
+        Set<PropertyInClass> props = new TreeSet<PropertyInClass>();
         for (Method m : clazz.getMethods()) {
-            if (m.getParameterTypes().length == 1) {
-                String propName = setter2property(m.getName());
-                if (propName != null) {
-                    props.add( new SetterInClass( propName, m.getDeclaringClass() ) );
+            if (m.getParameterTypes().length == 0) {
```
Comment (tarilabs) [inline drools-core/src/main/java/org/drools/core/util/ClassUtils.java:417]: iterating on the getters, means the 0-args methods.

File: drools-core/src/main/java/org/drools/core/util/ClassUtils.java [hunk 2]
```diff
@@ -411,13 +411,13 @@ public static String getPackage(Class<?> cls) {
...
+                String propName = getter2property(m.getName());
+                if (propName != null && !propName.equals( "class" )) {
```
Comment (tarilabs) [inline drools-core/src/main/java/org/drools/core/util/ClassUtils.java:419]: excluding the "class property", which is not really a property.

**Review by tarilabs — APPROVED** [review]
+1 now iterating generally on the getters for accessible properties, instead of before iterating directly on the setter.

**Pull request merged**